src/lib/rnp.c

```diff
@@ -1590,13 +1590,15 @@ rnp_decrypt_memory(
                           attempts,
                           &ctx->rnp->passphrase_provider);
     if (mem == NULL) {
-        return -1;
+        return RNP_ERROR_BAD_PARAMETERS;
     }
     // TODO: we should actually fail if output buffer is too small
```

Comment:
Doesn't it make more sense to just fail like this TODO mentions (rather than partially decrypt)?
(Also we're supposed to avoid in/out parameters like outsize is here.)

Reply:
Yes, this is actually a bug which I wanted to address in separated PR. But maybe quick and dirty fix here followed by real fix later is also good idea.